api/app.py

```diff
@@ -1,12 +1,96 @@
 from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy.sql import text
+from slackeventsapi import SlackEventAdapter
 import os
+from slack_sdk import WebClient
 
 app = Flask(__name__)
 
+app.config["SQLALCHEMY_DATABASE_URI"] = os.environ.get("DBHOST", "mysql://root:lolviper@localhost/Hogwarts")
+app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = True
+db = SQLAlchemy(app)
+
 @app.route("/")
 def home():
     return "Welcome to Hogwarts 🧙"
 
+@app.route('/test')
+def testdb():
+    try:
+        db.session.query("1").from_statement(text("SELECT 1")).all()
+        return '<h1>It works.</h1>'
+    except Exception as e:
+        print(e)
+        return '<h1>Something is broken.</h1>'
+
+# Bind the Events API route to your existing Flask app
+SLACK_SIGNING_SECRET=os.environ.get("SLACK_SIGNING_SECRET")
+slack_events_adapter = SlackEventAdapter(SLACK_SIGNING_SECRET, "/slack/events", app)
+
+# Create a SlackClient for your bot to use for Web API requests
+SLACK_BOT_TOKEN = os.environ["SLACK_BOT_TOKEN"]
+slack_client = WebClient(token=SLACK_BOT_TOKEN)
+
+SLACKBOT_USERID="U01F944MG3X"
+
+# Create an event listener for @bot mentions
+@slack_events_adapter.on("app_mention")
+def handle_app_mention(event_data):
+    message = event_data["event"]
+
+    # Get the user who initiated the @mention
+    source_user = slack_client.users_info(user=message["user"])
+
+    # Default variable values
+    slack_message, msg, mentioned_user = "", "", None
+    notify_user = source_user["user"]["id"]
+    
+    # Iterate through message body looking for the destination user. Use the first match.
+    elements = message["blocks"][0]["elements"][0]["elements"]
+    for element in elements:
+        if element["type"] == "user" and element["user_id"] != SLACKBOT_USERID:
+            mentioned_user = slack_client.users_info(user=element["user_id"])
+            break
+
+    # If there was a user mention, handle the points.
+    if mentioned_user is not None:
+        if source_user["user"]["id"] == mentioned_user["user"]["id"]:
+            msg = "Hey <@" + source_user["user"]["id"] + "> - nice try but you can only give points to others"
+        else:
+            # Try to get the number of points, default to 1.
+            text = message.get('text')
+            points_array = [int(s) for s in text.split() if s.isdigit()]
```

Comment:
Would it assign a number if there's ANY digit in the text then? Like if the text is "Kim created 2 endpoints"

Reply:
Yep, it would. There's definitely a more elegant solution out there, but I think this is probably adequate for an MVP 🙏